### PR TITLE
[tock] Add Tock UART tests.

### PR DESCRIPTION
**Pull request by jrvanwhy**
This test the ability of a Tock app to perform console writes and reads over the UART. It is an integration tests that tests the Console syscall driver, the UART virtualizer, and the UART peripheral driver.

### CODE REVIEW

File: sw/device/silicon_owner/tock/tests/uart/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    assert!(buffer.len() >= lens.iter().sum());
+    let upcall: Cell<Option<(u32, u32)>> = Cell::new(None);
+    let mut cursor = 0;
+    share::scope(|handle| {
+        TockSyscalls::subscribe::<_, _, DefaultConfig, 1, 2>(handle, &upcall).unwrap();
+        for &len in lens {
+            let bytes = share::scope(|handle| {
+                TockSyscalls::allow_rw::<DefaultConfig, 1, 1>(handle, &mut buffer[cursor..])
+                    .unwrap();
+                if let Some(error) = TockSyscalls::command(1, 2, len as u32, 0).get_failure() {
```
Comment (cfrantz) [inline sw/device/silicon_owner/tock/tests/uart/src/main.rs:30]: Are there symbolic constants for `driver_id` and `command_id` values?

File: sw/device/silicon_owner/tock/tests/uart/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+// Reads data from the console input. The data is read in chunks whose length is
+// given by `lens`. The newly-read data is returned. Panics upon encountering an
+// error condition.
+fn read_exact<'b>(buffer: &'b mut [u8], lens: &[usize]) -> &'b [u8] {
+    assert!(buffer.len() >= lens.iter().sum());
+    let upcall: Cell<Option<(u32, u32)>> = Cell::new(None);
+    let mut cursor = 0;
+    share::scope(|handle| {
+        TockSyscalls::subscribe::<_, _, DefaultConfig, 1, 2>(handle, &upcall).unwrap();
```
Comment (cfrantz) [inline sw/device/silicon_owner/tock/tests/uart/src/main.rs:25]: Similarly, are there symbolic constants that can be used in the `subscribe` and `allow_...` calls? (ie: these values are defined in `libtock-rs/apis/console/src/lib.rs` (although, they don't appear to be publicly exported from that module).

**Review by cfrantz — APPROVED** [review]
Just some comment nits.

File: sw/device/silicon_owner/tock/tests/uart/src/main.rs (hunk 1 above)
Comment (jrvanwhy) [reply sw/device/silicon_owner/tock/tests/uart/src/main.rs]: I added symbolic constants.

File: sw/device/silicon_owner/tock/tests/uart/src/main.rs (hunk 2 above)
Comment (jrvanwhy) [reply sw/device/silicon_owner/tock/tests/uart/src/main.rs]: There are not existing constants that can be used. I added new constants to the test and switched the code to use them.

**Pull request merged**